src/Compilers/CSharp/Test/Semantic/Semantics/NullCoalesceAssignmentTests.cs

```diff
@@ -186,5 +187,925 @@ void M(int? a)
+                        t ??= (new(a), 1);
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (6,20): error CS0103: The name 'a' does not exist in the current context
+                //         t ??= (new(a), 1);
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "a").WithArguments("a").WithLocation(6, 20));
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var descendantNodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var node = descendantNodes.OfType<TupleExpressionSyntax>().Single();
+
+            var typeInfo = model.GetTypeInfo(node);
+            Assert.Null(typeInfo.Type);
+            Assert.Equal("(C, System.Int32)", typeInfo.ConvertedType.ToTestDisplayString());
+
+            var assignmentNode = descendantNodes.OfType<AssignmentExpressionSyntax>().Single();
+            var assignmentTypeInfo = model.GetTypeInfo(assignmentNode);
+            Assert.Equal("(C, System.Int32)", assignmentTypeInfo.Type.ToTestDisplayString());
+            Assert.Equal("(C, System.Int32)", assignmentTypeInfo.ConvertedType.ToTestDisplayString());
+        }
+
+        [Theory]
+        [InlineData("=")]
+        [InlineData("??=")]
+        public void ErrorRecovery_TargetTypedConditionalExpression_ReferenceType(string assignmentOperator)
+        {
+            var source = $$"""
+                class C
+                {
+                    void M(bool b)
+                    {
+                        C c = null;
+                        c {{assignmentOperator}} b ? new(a) : default;
+                    }
+                }
+                """;
+
+            var comp = CreateCompilation(source);
+            comp.VerifyDiagnostics(
+                // (6,21): error CS0103: The name 'a' does not exist in the current context
+                //         c = b ? new(a) : default;
+                // (6,23): error CS0103: The name 'a' does not exist in the current context
+                //         c ??= b ? new(a) : default;
+                Diagnostic(ErrorCode.ERR_NameNotInContext, "a").WithArguments("a").WithLocation(6, 20 + assignmentOperator.Length));
+
+            var tree = comp.SyntaxTrees.First();
+            var model = comp.GetSemanticModel(tree);
+
+            var descendantNodes = tree.GetCompilationUnitRoot().DescendantNodes();
+            var node = descendantNodes.OfType<ConditionalExpressionSyntax>().Single();
+
+            var typeInfo = model.GetTypeInfo(node);
+            // INCONSISTENCY: if conditional expression didn't have any errors, it would have a `null` type.
+            // Shouldn't this be preserved given that an expression is target-typed and have no natural type anyway?
+            // Moreover, target-typed tuples and collection expressions already behave that way
```

Comment:
Error cases do error things. Having them be fully consistent with non-error scenarios is nice, but ultimately not a requirement.

Reply:
Sooo, do we just left everything as is and I should just delete all these comments?

Reply:
For the cases that make sense to do so, yes. Let's get the `new T()` change merged, then come back and update all of these baselines and see where we're at.